org-code-javabuilder/protocol/src/main/java/org/code/protocol/GlobalProtocol.java

```diff
@@ -0,0 +1,32 @@
+package org.code.protocol;
+
+public class GlobalProtocol {
+  private static GlobalProtocol protocolInstance;
+  private final OutputAdapter outputAdapter;
+  private final InputAdapter inputAdapter;
+
+  private GlobalProtocol(OutputAdapter outputAdapter, InputAdapter inputAdapter) {
+    this.outputAdapter = outputAdapter;
+    this.inputAdapter = inputAdapter;
+  }
+
+  public static void create(OutputAdapter outputAdapter, InputAdapter inputAdapter) {
+    GlobalProtocol.protocolInstance = new GlobalProtocol(outputAdapter, inputAdapter);
```

Comment:
should this throw an error if there is already a protocolInstance?

Reply:
That's a good question - for the local scenario or the lambda scenario there's no valid instance where GlobalProtocol.protocolInstance would already exist. But for the local websocket server scenario, it would i.e. the instance gets set onOpen, but never gets removed until the server is turned off.

Reply:
Now that I'm writing this out, there is an opportunity to unset the instance onClose. Although thinking through that, we might run into problems running javabuilder in multiple tabs. But it's also only for a local development scenario.

Reply:
I'm not sure there's an immediate good answer here. Happy to take suggestions. For now, I'm leaving as-is

Reply:
My general thought here (non-blocking) is a singleton should only have 1 instance of the object. Allowing create to overwrite it seems like an anti-pattern. If we want it to be able to be overwritten for certain cases that's ok, but if we don't then maybe throw an exception or just have this be a no-op?

Reply:
Added a task to track this. I'll try out a couple things. For now, merging to unblock Theater development.

Reply:
https://codedotorg.atlassian.net/browse/CSA-445